gnocchi/rest/__init__.py

```diff
@@ -535,21 +535,41 @@ def MetricSchema(cls, definition):
         return definition
 
     @pecan.expose('json')
-    def post(self):
+    def post(self, source=None):
         creator = pecan.request.auth_helper.get_current_user(
             pecan.request)
         body = deserialize_and_validate(self.MetricSchema)
-        try:
-            m = pecan.request.indexer.create_metric(
-                uuid.uuid4(),
-                creator,
-                name=body.get('name'),
-                unit=body.get('unit'),
-                archive_policy_name=body['archive_policy_name'])
-        except indexer.NoSuchArchivePolicy as e:
-            abort(400, e)
+
+        if not source:
+            try:
+                m = pecan.request.indexer.create_metric(
+                    uuid.uuid4(),
+                    creator,
+                    name=body.get('name'),
+                    unit=body.get('unit'),
+                    archive_policy_name=body['archive_policy_name'])
+            except indexer.NoSuchArchivePolicy as e:
+                abort(400, e)
+        else:
+            source_metric = pecan.request.indexer.list_metrics(
+                id=source, details=True)[0]
```

Comment:
Is there a better way for obtaining a single metric in this case? I do not like the idea of using `list_metrics`. I couldn't find a method for this in the Metric controller, I am thinking the `_lookup` method is handling that somehow.